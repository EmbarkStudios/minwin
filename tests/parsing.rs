use std::fmt::Write;

#[test]
fn parses_simple() {
    let mut parser = minwin::Parser::default();
    parser.add_file("tests/data/parking_lot.rs");

    let parsed = parser.parse();

    let mut actual = String::new();

    for pf in parsed {
        writeln!(&mut actual, "{}", pf.path).unwrap();

        let mut module = None;

        for i in pf.iter_binds() {
            if Some(&i.module.ident) != module {
                writeln!(&mut actual, "mod {}", i.module.ident).unwrap();
                module = Some(&i.module.ident);
            }
            writeln!(&mut actual, "    {:?} {}", i.kind, i.ident).unwrap();
        }
    }

    insta::assert_snapshot!(actual);
}
