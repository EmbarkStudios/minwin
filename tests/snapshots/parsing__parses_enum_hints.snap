---
source: tests/parsing.rs
expression: actual
---
tests/data/enums.rs
mod enums
    Enum WIN32_ERROR @ None
    Function GetLastError @ None
    Enum DUPLICATE_HANDLE_OPTIONS @ None
    Enum FILE_NOTIFY_CHANGE @ None
    Enum ALG_ @ Some(u!("Windows.Win32.Security.Cryptography"))
    Enum _PROP_ID @ Some(u!("Windows.Win32.Security.Cryptography"))

